packages/installations/src/helpers/extract-app-config.ts

```diff
@@ -15,21 +15,44 @@
  * limitations under the License.
  */
 
-import { FirebaseApp } from '@firebase/app-types';
+import { FirebaseApp, FirebaseOptions } from '@firebase/app-types';
 import { AppConfig } from '../interfaces/app-config';
 import { ERROR_FACTORY, ErrorCode } from '../util/errors';
+import { FirebaseError } from '@firebase/util';
 
 export function extractAppConfig(app: FirebaseApp): AppConfig {
   if (!app || !app.options) {
-    throw ERROR_FACTORY.create(ErrorCode.MISSING_APP_CONFIG_VALUES);
+    throw getMissingValueError('App Configuration');
   }
 
-  const appName = app.name;
-  const { projectId, apiKey, appId } = app.options;
+  if (!app.name) {
+    throw getMissingValueError('App Name');
+  }
+
+  // Required app config keys
+  const configKeys: Array<keyof FirebaseOptions> = [
+    'projectId',
+    'apiKey',
+    'appId'
+  ];
 
-  if (!appName || !projectId || !apiKey || !appId) {
-    throw ERROR_FACTORY.create(ErrorCode.MISSING_APP_CONFIG_VALUES);
+  for (const keyName of configKeys) {
+    if (!app.options[keyName]) {
+      // Cast required becase FirebaseOptions includes "[name: string]: any;".
```

Comment:
:/

Reply:
Be the change you want to see in the world: #2298.

Reply:
👏🎉